plugins/forms/dialogs/checkbox.js

```diff
@@ -138,7 +138,8 @@ CKEDITOR.dialog.add( 'checkbox', function( editor ) {
 				accessKey: 'Q',
 				value: 'required',
 				setup: function( element ) {
-					this.setValue( element.getAttribute( 'required' ) );
+					var required = element.getAttribute( 'required' );
+					this.setValue( required === '' || required === 'required' );
```

Comment:
This code should be extracted to some helper as it is repeated 5 times.

We could even introduce more general approach which will take into consideration all Boolean Attributes (such attributes are true when just present in an element), e.g. in `getAttribute` method (here is a list of Boolean Attributes, not sure if not outdated https://github.com/kangax/html-minifier/issues/63#issuecomment-18634279).